### PR TITLE
Fix create_bwc_index for 5.x

### LINKED ISSUE #19253: Create-bwc index tool fails with mapping exception
Trying to create a bwc index for 5.0.0-alpha4  with

    python3 dev-tools/create_bwc_index.py 5.0.0-alpha4

fails with a mapping exception:

    Traceback (most recent call last):
      File "dev-tools/create_bwc_index.py", line 461, in <module>
        main()
      File "dev-tools/create_bwc_index.py", line 457, in main
        create_bwc_index(cfg, version)
      File "dev-tools/create_bwc_index.py", line 421, in create_bwc_index
        generate_index(client, version, index_name)
      File "dev-tools/create_bwc_index.py", line 300, in generate_index
        'warmers': warmers
      File "/usr/local/lib/python3.5/site-packages/elasticsearch/client/utils.py", line 69, in _wrapped
        return func(*args, params=params, **kwargs)
      File "/usr/local/lib/python3.5/site-packages/elasticsearch/client/indices.py", line 103, in create
        params=params, body=body)
      File "/usr/local/lib/python3.5/site-packages/elasticsearch/transport.py", line 307, in perform_request
        status, headers, data = connection.perform_request(method, url, params, body, ignore=ignore, timeout=timeout)
      File "/usr/local/lib/python3.5/site-packages/elasticsearch/connection/http_urllib3.py", line 93, in perform_request
        self._raise_error(response.status, raw_data)
      File "/usr/local/lib/python3.5/site-packages/elasticsearch/connection/base.py", line 105, in _raise_error
        raise HTTP_EXCEPTIONS.get(status_code, TransportError)(status_code, error_message, additional_info)
    elasticsearch.exceptions.RequestError: TransportError(400, 'mapper_parsing_exception')

### PR DESCRIPTION
**Pull request by s1monw**
`create_bwc_index.py` has some cruft in it that only works on 2.x or
even before. This commit make the tool functional, yet there might still
be some bwc relevant things missing here.

Closes #19253

### CODE REVIEW

File: dev-tools/create_bwc_index.py [hunk 1]
```diff
@@ -250,13 +250,13 @@ def generate_index(client, version, index_name):
   mappings['norms'] = {
     'properties': {
       'string_with_norms_disabled': {
-        'type': 'string',
+        'type': 'text',
         'norms': {
           'enabled': False
         }
```
Comment (jpountz) [inline dev-tools/create_bwc_index.py]: should be `'norms': False`

File: dev-tools/create_bwc_index.py [hunk 2]
```diff
@@ -250,13 +250,13 @@ def generate_index(client, version, index_name):
...
       },
       'string_with_norms_enabled': {
-        'type': 'string',
+        'type': 'text',
         'index': 'not_analyzed',
         'norms': {
           'enabled': True,
```
Comment (jpountz) [inline dev-tools/create_bwc_index.py]: should be `'norms': True`

**Comment by jpountz** [comment]
LGTM

File: dev-tools/create_bwc_index.py (hunk 1 above)
Comment (s1monw) [inline dev-tools/create_bwc_index.py]: ok but it still works?

**Comment by s1monw** [comment]
@jpountz pushed more changes

**Comment by jpountz** [comment]
LGTM2

**Pull request merged**